src/image/loading_displaying.js

```diff
@@ -158,6 +159,363 @@ p5.prototype.loadImage = function(path, successCallback, failureCallback) {
+ * animation to be, if you are very sure of this number.
+ *
+ * @method saveGif
+ * @param  {String} filename File name of your gif
+ * @param  {Number} duration Duration in seconds that you wish to capture from your sketch
+ * @param  {Object} options An optional object that can contain two more arguments: delay, specifying 
+ * how much time we should wait before recording, and units, a string that can be either 'seconds' or 
+ * 'frames'. By default it's 'seconds'. 
+ *
+ * @example
+ * <div>
+ * <code>
+ * function setup() {
+ *   createCanvas(100, 100);
+ *   colorMode(HSL);
+ * }
+
+ * function draw() {
+ *   // create some cool dynamic background
+ *   let hue = map(sin(frameCount / 100), -1, 1, 0, 100);
+ *   background(hue, 40, 60);
+
+ *   // create a circle that moves diagonally
+ *   circle(
+ *     100 * sin(frameCount / 10) + width / 2,
+ *     100 * sin(frameCount / 10) + height / 2,
+ *     10
+ *   );
+ * }
+
+ * // you can put it in the mousePressed function,
+ * // or keyPressed for example
+ * function mousePressed() {
+ *   // this will download the first two seconds of my animation!
+ *   saveGif('mySketch', 2, { units: 'seconds', delay: 0 });
+ * }
+ * </code>
+ * </div>
+ *
+ * @alt
+ * animation of a circle moving smoothly diagonally
+ */
+p5.prototype.saveGif = async function(
+  fileName,
+  duration,
+  { delay = 0, units = 'seconds' }
+) {
+  // validate parameters to throw friendly error
+  p5._validateParameters('saveGif', arguments);
+
+  // get the project's framerate
+  let _frameRate = this._targetFrameRate;
+  // if it is undefined or some non useful value, assume it's 60
+  if (_frameRate === Infinity || _frameRate === undefined || _frameRate === 0) {
+    _frameRate = 60;
+  }
+
+  // calculate frame delay based on frameRate
+
+  // this delay has nothing to do with the
+  // delay in options, but rather is the delay
+  // we have to specify to the gif encoder between frames.
+  let gifFrameDelay = 1 / _frameRate * 1000;
+
+  // constrain it to be always greater than 20,
+  // otherwise it won't work in some browsers and systems
+  // reference: https://stackoverflow.com/questions/64473278/gif-frame-duration-seems-slower-than-expected
+  gifFrameDelay = gifFrameDelay < 20 ? 20 : gifFrameDelay;
+
+  // check the mode we are in and how many frames
+  // that duration translates to
+  const nFrames = units === 'seconds' ? duration * _frameRate : duration;
+  const nFramesDelay = units === 'seconds' ? delay * _frameRate : delay;
+
+  // initialize variables for the frames processing
+  let count = nFramesDelay;
+
+  // we start on the frame set by the delay argument
+  //   frameCount = nFramesDelay;
+
+  const lastPixelDensity = this._pixelDensity;
+  this.pixelDensity(1);
+
+  // We first take every frame that we are going to use for the animation
+  let frames = [];
+
+  if (document.getElementById('progressBar') !== null)
```

Comment:
progressBar sounds so common name, lets name it something that would be unique that perhaps user wont be using .


something like - p5.gif.progressBar

Reply:
Sounds great, will fix it ASAP!